addons/web_editor/static/src/js/editor/odoo-editor/src/OdooEditor.js

```diff
@@ -1971,8 +1971,15 @@ export class OdooEditor extends EventTarget {
         // Let the DOM split and delete the range.
         const contents = range.extractContents();
 
-        setSelection(start, nodeSize(start));
         const startLi = closestElement(start, 'li');
+        const endli = closestElement(end, 'li');
+        let shouldMergeLi;
+        if (endli && startLi && isZWS(startLi) && getListMode(startLi.parentElement) !== getListMode(endli.parentElement)) {
```

Comment:
With this you will also have to make sure `doJoin` is false for this case. Also add a test case for it:
`<ul><li>abc</li><li>[def</li></ul><ol><li>g]hi</li><li>jkl</li></ol>` => `deleteBackward`.